src/main/java/uk/gov/hmcts/reform/et/syaapi/service/AssignCaseToLocalOfficeService.java

```diff
@@ -0,0 +1,67 @@
+import org.springframework.util.CollectionUtils;
+import uk.gov.dwp.regex.InvalidPostcodeException;
+import uk.gov.hmcts.ecm.common.model.helper.TribunalOffice;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.reform.et.syaapi.helper.EmployeeObjectMapper;
+import uk.gov.hmcts.reform.et.syaapi.models.CaseRequest;
+
+import java.util.List;
+
+import static uk.gov.hmcts.ecm.common.model.helper.TribunalOffice.getCaseTypeId;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.DEFAULT_TRIBUNAL_OFFICE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.UNASSIGNED_OFFICE;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AssignCaseToLocalOfficeService {
+    private final PostcodeToOfficeService postcodeToOfficeService;
+
+    /**
+     * Assigns case to local office. Returns @{@link CaseData} object.
+     *
+     * @param caseRequest which would be in json format
+     * @return @link CaseData
+     */
+    public CaseData convertCaseRequestToCaseDataWithTribunalOffice(CaseRequest caseRequest) {
+        CaseData caseData = EmployeeObjectMapper.mapRequestCaseDataToCaseData(caseRequest.getCaseData());
+        List<RespondentSumTypeItem> respondentSumTypeList = caseData.getRespondentCollection();
+        String managingOffice = UNASSIGNED_OFFICE;
+        if (caseData.getClaimantWorkAddress() != null
+            && caseData.getClaimantWorkAddress().getClaimantWorkAddress() != null
+            && StringUtils.isNotBlank(caseData.getClaimantWorkAddress().getClaimantWorkAddress().getPostCode())) {
+            managingOffice = getManagingOffice(
+                caseData.getClaimantWorkAddress().getClaimantWorkAddress().getPostCode());
+        } else if (!CollectionUtils.isEmpty(respondentSumTypeList)) {
+            for (RespondentSumTypeItem respondentSumTypeItem : respondentSumTypeList) {
+                if (respondentSumTypeItem.getValue() != null
+                    && respondentSumTypeItem.getValue().getRespondentAddress() != null
+                    && StringUtils.isNotBlank(respondentSumTypeItem.getValue().getRespondentAddress().getPostCode())) {
+                    managingOffice = getManagingOffice(
+                        respondentSumTypeItem.getValue().getRespondentAddress().getPostCode());
+                    break;
+                }
+            }
+        }
+        caseData.setManagingOffice(managingOffice);
+        return caseData;
+    }
+
+    private String getManagingOffice(String postcode) {
+        try {
+            return postcodeToOfficeService.getTribunalOfficeFromPostcode(postcode)
+                .map(TribunalOffice::getOfficeName)
+                .orElse(UNASSIGNED_OFFICE);
+        } catch (InvalidPostcodeException e) {
+            log.info("Failed to find tribunal office : {} ", e.getMessage());
+            return getCaseTypeId(DEFAULT_TRIBUNAL_OFFICE.getOfficeName());
```

Comment:
This should be unassigned too shouldn't it? I think we should remove that Default Office as London South

Reply:
Not sure if we can have 3 types of Id's. I think we talked about it with Mehmet and Paul. Like It has to be ET_EnglandWales and ET_Scotland, or I missunderstood something?

Reply:
Fixed that